native/src/main/java/io/ballerina/stdlib/http/api/ValueCreatorUtils.java

```diff
@@ -109,19 +109,17 @@ public static BMap<BString, Object> createHTTPRecordValue(String recordTypeName)
      * @param fieldValues values to be used for fields when creating the object value instance.
      * @return value of the object.
      */
-    private static BObject createObjectValue(Module module, String objectTypeName,
-                                             Object... fieldValues) {
-
-        Object[] fields = new Object[fieldValues.length * 2];
-
-        // Adding boolean values for each arg
-        for (int i = 0, j = 0; i < fieldValues.length; i++) {
-            fields[j++] = fieldValues[i];
-            fields[j++] = true;
+    private static BObject createObjectValue(Module module, String objectTypeName, Object... fieldValues) {
```

Comment:
Is this change needed? Even if we fix this from http, runtime does not support this right?

Reply:
There was an issue in how this was implemented. Earlier, we have passed `ValueCreator.createObjectValue(module, objectTypeName, null, null, null, fields);` mistakenly. But the `ValueCreator` API is `public static BObject createObjectValue(Module packageId, String objectTypeName, Object... fieldValues) {`